### ISSUE TITLE
Postman 8.0 Runner performance issue

### ISSUE THREAD

**Issue by ManiacDC** [a85/POSTMan-Chrome-Extension#9476]
**Describe the Issue**

After running my collection, Postman froze up for a couple minutes. It then unfroze, but started responding very slowly. Memory usage was almost 2GB. About 5 minutes after that, it started running normally and memory usage dropped down to about 1GB. My collection has about 1000 requests and over 2.5k assertions. I did have save responses checked off.

This did not happen with prior versions of Postman.

**Steps to Reproduce**

Run a large collection, Postman will become unresponsive.

**Screenshots or Videos**
![image](https://user-images.githubusercontent.com/1823305/106045184-a3959f80-60ae-11eb-9ee1-dd348247a61d.png)

**Environment Information**

 - Platform Type: Native Desktop App
 - Postman Version: 8.0.2
 - Operating System: Windows 10

**Comment by ManiacDC** [comment]
Also noticed today that after running a collection, CPU usage seems to never drop:
![image](https://user-images.githubusercontent.com/1823305/106153426-88796d00-614c-11eb-9d81-a576ed839cb5.png)

Even after closing the runner tab:
![image](https://user-images.githubusercontent.com/1823305/106153583-b2cb2a80-614c-11eb-8760-13356e5c4a9a.png)

**Comment by mstaalesen** [comment]
Can you check if Postman keeps logging to Postman Console even after the collection has executed all the requests? I believe that is the issue for me at least.

**Comment by ManiacDC** [comment]
@mstaalesen that appears to be the case for me as well. Nice find.

**Comment by ManiacDC** [comment]
Yep, after the console *finally* finished logging everything, the CPU usage went down. It appears the console is lagging way behind the actual requests.

**Comment by mstaalesen** [comment]
I created a feature-request to have a way of disabling the logging: https://github.com/postmanlabs/postman-app-support/issues/9544#issuecomment-774499873

**Comment by malvikach** [comment]
Update: We have released some Performance Improvements in Collection Runner in Postman version. 10.9.3. This should improve memory usage by runner (by freeing up memory during the run).

Our internal benchmarking suggests that this should improve the usage by 2x( the same amount of memory to be consumed for double the no.of iterations or double the size of the collection)

Please let us know your feedback on the above and if this helps your workflows.

We are trying to gather more information on the performance challenges that you face, please feel free to setup some time with me using [this](https://calendly.com/malvika-chaudhary/v10-feedback) calendly link, if you would like to share more information or walk us through your usecase.

Note: We are evaluating having the option to suppress console logs too meanwhile. Will share more updates on the https://github.com/postmanlabs/postman-app-support/issues/9476#:~:text=May%2020%2C%202021-,Postman%20console%20lags%20behind%20Collection%20Runner%20on%20Desktop%20Application%20%239929,-Open regarding this.

**Comment by NelemaniA** [comment]
feature-request 9544 as mentioned above is finally implemented and might solve this problem as well.

**Comment by malvika-chaudhary** [comment]
Hi all 👋 

Glad to inform you that we have made some Performance Improvements to collection runner to reduce the CPU utilization/overhead especially while running large collections.

Please let us know if you still see any issues while running large collections. I will be closing this ticket, but please feel free to reopen if your issues still persists.


Cheers!
Malvika

**Comment by NelemaniA** [comment]
@malvika-chaudhary Starting from which release is this feature implemented?

**Comment by saksham-postman** [comment]
@NelemaniA you will be able to find this feature from version 11.29.5 onwards
